services/QuillLMS/app/services/classlink_integration/school_persister.rb

```diff
@@ -27,8 +31,8 @@ def run
     end
 
     private def school
-      school = nces_id ? School.find_by(nces_id:, district_id:) : School.find_by(name:, district_id:)
-      school.tap { |s| s.update(name:) }
+      found_school = (nces_id && School.find_by(nces_id:, district_id:)) || School.find_by(name:, district_id:)
```

Comment:
This feels slightly hard to parse.  I wonder if it's worth breaking this into a pair of sub-methods:
```ruby
private def school_by_nces_id
  return unless nces_id

  School.find_by(nces_id: district_id:)
end

private def school_by_name = School.find_by(name:, district_id:)

private def school
  found_school = school_by_nces_id || school_by_name
  found_school if premium_school_ids.include?(found_school&.id)
end
```